python/src/deltachat/account.py

```diff
@@ -333,12 +338,22 @@ def stop_threads(self, wait=True):
         lib.dc_stop_ongoing_process(self._dc_context)
         self._threads.stop(wait=wait)
 
+    def shutdown(self, wait=True):
+        """ stop threads and close and remove underlying dc_context and callbacks. """
+        if hasattr(self, "_dc_context"):
+            self.stop_threads(wait=False)  # to interrupt idle and tell python threads to stop
+            lib.dc_close(self._dc_context)
+            self.stop_threads(wait=wait)  # to wait for threads
+            deltachat.clear_context_callback(self._dc_context)
```

Comment:
only hear removing the callback -- we want to get all events from the termination phase.